doc/debugger.rst

```diff
@@ -1773,6 +1808,74 @@ target endianness and interpret the bytecode based on that.
 .. note:: This command is somewhat incomplete at the moment and may be modified
    once the best way to do this in the debugger UI has been figured out.
 
+AppRequest request (0x23)
+-------------------------
+
+Format::
+
+    REQ <int: 0x23> [<tval>*] EOM
+    REP [<tval>*] EOM
+
+Example::
+
+    REQ 35 "GameInfo" "GetTitle" EOM
+    REP "Spectacles: Bruce's Story" EOM
+
+This is a custom request message whose meaning and semantics depend on the
+application.
+
+AppRequest messages are used for direct communication between the debug client
+and debug target over the Duktape debug protocol. Both the meaning of a custom
+message and the dvalues it contains are entirely up to the implementation and
+depending on the needs of the application, need not be supported at all.
+
+See "Custom requests and notifications" below for more details.
+
+Custom requests and notifications
+=================================
+
+Starting in Duktape 1.5.x, Duktape supports direct communication between the
+debug client and debug target over the same transport by using the special
+'AppRequest' and 'AppNotify' messages. These messages have no meaning to
+Duktape, which merely serves to marshal them back and forth through a defined
+API.
+
+AppNotify messages may be sent by pushing the contents of the message to the
+stack and calling `duk_debugger_notify()` passing the number of values pushed.
+Each value pushed will be sent as a dvalue in the message. So if you push two
+strings, "foo" and "bar", the client will see ``NFY 7 "foo" "bar" EOM``.
+
+AppRequests are used to make requests to the target which are not directly
+related to Ecmascript execution. For example, a game engine might allow the
+debug client to change the framerate, or request information about the
+currently running game. Supporting this requires attaching using the
+``duk_debugger_attach_custom()`` API and providing a request callback. When an
```

Comment:
Hmm, maybe the usage example should be moved into bullets so that they'd pop out a little more.

I'd actually like to also mention (in addition to the reset):

- Performing or triggering software or script updates

A few very different examples would be nice because this is a pretty flexible feature which can be used for a lot of things. Some targets will have often means of communication, but it'd actually be awesome if a debug client had custom support for e.g. fixing source code in the debugger directly, and using a custom command to update the script on the target, and then to restart execution - all without exiting the debugger.